support-frontend/app/controllers/WeeklySubscription.scala

```diff
@@ -0,0 +1,89 @@
+package controllers
+
+import actions.CustomActionBuilders
+import admin.settings.{AllSettings, AllSettingsProvider, SettingsSurrogateKeySyntax}
+import assets.{AssetsResolver, RefPath, StyleContent}
+import cats.implicits._
+import com.gu.googleauth.AuthAction
+import com.gu.identity.play.IdUser
+import com.gu.monitoring.SafeLogger
+import com.gu.monitoring.SafeLogger._
+import com.gu.support.catalog.Paper
+import com.gu.support.config.{PayPalConfigProvider, StripeConfigProvider}
+import com.gu.support.encoding.CustomCodecs._
+import com.gu.support.pricing.PriceSummaryServiceProvider
+import config.StringsConfig
+import play.api.libs.circe.Circe
+import play.api.mvc._
+import play.twirl.api.Html
+import services.{IdentityService, TestUserService}
+import views.EmptyDiv
+import views.ViewHelpers.outputJson
+import views.html.helper.CSRF
+import views.html.subscriptionCheckout
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class WeeklySubscription(
+  authAction: AuthAction[AnyContent],
+  priceSummaryServiceProvider: PriceSummaryServiceProvider,
+  val assets: AssetsResolver,
+  val actionRefiners: CustomActionBuilders,
+  identityService: IdentityService,
+  testUsers: TestUserService,
+  stripeConfigProvider: StripeConfigProvider,
+  payPalConfigProvider: PayPalConfigProvider,
+  components: ControllerComponents,
+  stringsConfig: StringsConfig,
+  settingsProvider: AllSettingsProvider,
+  val supportUrl: String,
+  fontLoaderBundle: Either[RefPath, StyleContent]
+)(implicit val ec: ExecutionContext) extends AbstractController(components) with GeoRedirect with Circe with CanonicalLinks with SettingsSurrogateKeySyntax {
+
+  import actionRefiners._
+
+  implicit val a: AssetsResolver = assets
+
+
+  def displayForm(): Action[AnyContent] =
+    (PrivateAction andThen authAction andThen authenticatedAction(subscriptionsClientId)).async { implicit request =>
```

Comment:
oh yeah I also added google auth to it to prevent it being discovered by search engines before I remembered we were going to use a URL parameter originally.
The reason we were suggesting that is so we didn't forget to remove it before going live.  So I will add something obvious in the URL before shipping this.

Reply:
This is fine, but do put a big TODO or something in here so we don't forget to remove it